### PR TITLE
checkpoint naming in nonray mode, fix ray mode, delete checkpoints in nonray mode

### PR DESCRIPTION
**Pull request by liususan091219**
…ning test mode, delete all the checkpoints in non-ray mode

### CODE REVIEW

File: flaml/model.py [hunk 1]
```diff
@@ -513,10 +552,16 @@ def _compute_metrics_by_dataset_name(self, eval_pred):
             if self._task == SEQREGRESSION
             else np.argmax(predictions, axis=1)
         )
+        from .nlp.utils import load_default_huggingface_metric_for_task
+        import datasets
+        default_metric_name = load_default_huggingface_metric_for_task(self._task)
+        metric = datasets.load_metric(default_metric_name)
```
Comment (sonichi) [inline flaml/model.py:558]: move into "else"

File: flaml/model.py [hunk 2]
```diff
@@ -513,10 +552,16 @@ def _compute_metrics_by_dataset_name(self, eval_pred):
...
+        import datasets
+        default_metric_name = load_default_huggingface_metric_for_task(self._task)
+        metric = datasets.load_metric(default_metric_name)
         return {
-            self._metric_name: sklearn_metric_loss_score(
+            "val_loss": sklearn_metric_loss_score(
                 metric_name=self._metric_name, y_predict=predictions, y_true=labels
-            )
+            ) } \
+            if isinstance(self._metric_name, str) else {
```
Comment (sonichi) [inline flaml/model.py:563]: lint

File: flaml/nlp/huggingface/trainer.py [hunk 3]
```diff
@@ -11,9 +11,12 @@ def evaluate(self, eval_dataset=None, ignore_keys=None, metric_key_prefix="eval"
         """Overriding transformers.Trainer.evaluate by saving metrics and checkpoint path"""
         from transformers.trainer_utils import PREFIX_CHECKPOINT_DIR
 
-        ckpt_dir = os.path.join(
+        if self._use_ray is False:
+            ckpt_dir = os.path.join(
             self.args.output_dir, f"{PREFIX_CHECKPOINT_DIR}-{self.state.global_step}"
         )
```
Comment (sonichi) [inline flaml/nlp/huggingface/trainer.py:17]: lint

File: flaml/nlp/huggingface/trainer.py [hunk 4]
```diff
@@ -11,9 +11,12 @@ def evaluate(self, eval_dataset=None, ignore_keys=None, metric_key_prefix="eval"
...
+        else:
+            ckpt_dir = os.path.join(self.args.output_dir, f"{PREFIX_CHECKPOINT_DIR}-{self.state.global_step}")
```
Comment (sonichi) [inline flaml/nlp/huggingface/trainer.py:19]: what's the difference between if and else?

File: test/nlp/tmp.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
+import os
```
Comment (sonichi) [inline test/nlp/tmp.py:1]: what's the purpose of this file?

**Review by sonichi — COMMENTED** [review]
first round of review.

File: flaml/model.py [hunk 6]
```diff
@@ -86,6 +87,9 @@ def __init__(self, task="binary", **config):
             config: A dictionary containing the hyperparameter names, 'n_jobs' as keys.
                 n_jobs is the number of parallel threads.
         """
+        import uuid
+
+        self.trial_id = str(uuid.uuid1().hex)[:8]
```
Comment (sonichi) [inline flaml/model.py:92]: BaseEstimator doesn't need this. Move to TransformerEstimator

File: flaml/model.py [hunk 7]
```diff
@@ -342,10 +351,15 @@ def _preprocess(self, X, task, **kwargs):
...
     def fit(self, X_train: DataFrame, y_train: Series, budget=None, **kwargs):
+
         # TODO: when self.param = {}, ie max_iter = 1, fix the bug
         from transformers import EarlyStoppingCallback
+        import sys
 
         this_params = self.params
+        this_params[TransformersEstimator.ITER_HP] = this_params.get(
+            TransformersEstimator.ITER_HP, sys.maxsize
+        )
```
Comment (sonichi) [inline flaml/model.py:362]: move to self.config2params()

File: flaml/model.py [hunk 8]
```diff
@@ -513,11 +570,24 @@ def _compute_metrics_by_dataset_name(self, eval_pred):
...
+            }
+        else:
+            from .nlp.utils import load_default_huggingface_metric_for_task
+            import datasets
+
+            default_metric_name = load_default_huggingface_metric_for_task(self._task)
+            metric = datasets.load_metric(default_metric_name)
+            return {
+                "val_loss": metric.compute(predictions=predictions, references=labels)[
+                    default_metric_name
```
Comment (sonichi) [inline flaml/model.py:588]: Is the default metric lower the better? If not, need to use minus.

File: test/nlp/test_autohf_maxiter1.py [hunk 9]
```diff
@@ -0,0 +1,68 @@
+import os
+import pytest
+
+
+@pytest.mark.skipif(os.name == "posix", reason="do not run on mac os")
+def test_max_iter_1():
+    try:
+        import ray
+    except ImportError:
+        return
```
Comment (sonichi) [inline test/nlp/test_autohf_maxiter1.py:10]: Does this test require ray?

File: test/nlp/test_autohf_ray.py [hunk 10]
```diff
@@ -0,0 +1,52 @@
+import os
```
Comment (sonichi) [inline test/nlp/test_autohf_ray.py:1]: I suggest changing an existing test into use ray. Any drawback for that?

**Review by sonichi — CHANGES_REQUESTED** [review]

File: flaml/model.py [hunk 11]
```diff
@@ -578,23 +567,47 @@ def _compute_metrics_by_dataset_name(self, eval_pred):
...
+                    "val_loss": -metric.compute(
+                        predictions=predictions, references=labels
+                    )[default_metric_name]
+                }
+            else:
+                return {
+                    "val_loss": metric.compute(
+                        predictions=predictions, references=labels
+                    )[default_metric_name]
+                }
```
Comment (sonichi) [inline flaml/model.py:586]:
```suggestion
            multiplier = -1 if default_metric_mode == "max" else 1
            return {
                "val_loss": metric.compute(
                    predictions=predictions, references=labels
                )[default_metric_name] * multiplier
            }

```

File: flaml/model.py [hunk 12]
```diff
@@ -578,23 +567,47 @@ def _compute_metrics_by_dataset_name(self, eval_pred):
...
-        from .nlp.utils import load_model
-        from transformers import TrainingArguments
-        from .nlp.huggingface.trainer import TrainerForAuto
+    def predict_proba(self, X_test):
+        assert (
+            self._task in CLASSIFICATION
+        ), "predict_proba is only available in classification tasks"
+        if X_test.dtypes[0] == "string":
+            X_test = self._preprocess(X_test, self._task, **self._kwargs)
+            test_dataset = Dataset.from_pandas(X_test)
```
Comment (sonichi) [inline flaml/model.py]: This doesn't look right.
line 592 and 594 should be inside _preprocess().
Make sure test_dataset is defined in all cases as it is needed in line 607.

File: flaml/model.py [hunk 13]
```diff
@@ -578,23 +567,47 @@ def _compute_metrics_by_dataset_name(self, eval_pred):
...
+            output_dir=self.custom_hpo_args.output_dir,
+        )
+        self._model = TrainerForAuto(model=best_model, args=training_args)
+        predictions = self._model.predict(test_dataset)
+        return predictions.predictions
+
+    def predict(self, X_test):
         if X_test.dtypes[0] == "string":
             X_test = self._preprocess(X_test, self._task, **self._kwargs)
             test_dataset = Dataset.from_pandas(X_test)
```
Comment (sonichi) [inline flaml/model.py]: same problem

**Review by sonichi — CHANGES_REQUESTED** [review]
please link relevant issues.

**Comment by liususan091219** [comment]
Closes Issue #290 #291

File: flaml/automl.py [hunk 14]
```diff
@@ -739,6 +739,8 @@ def _preprocess(self, X):
                 X = DataFrame(
                     {
                         self._transformer._str_columns[idx]: X[idx]
+                        if isinstance(X[0], List)
+                        else [X[idx]]
```
Comment (sonichi) [inline flaml/automl.py:743]: why not
```suggestion
                        else self._transformer._str_columns[idx]: [X[idx]]
```

**Review by sonichi — APPROVED** [review]

**Review by sonichi — COMMENTED** [review]
remove 
NLP
---

.. autoclass:: flaml.nlp.HPOArgs
   :members:
   
from index.rst?

**Pull request merged**